examples/99_in_tool_selection/ex_in_tool_selection/main.rb

```diff
@@ -0,0 +1,364 @@
+          return
+        end
+        direction = picked_points[1] - picked_points[0]
+        end_point = picked_points[0].offset(direction, distance)
+        @mouse_ip = Sketchup::InputPoint.new(end_point)
+
+        do_move(view)
+        end_move(view)
+
+        view.invalidate
+      end
+
+      def draw(view)
+        draw_preview_line(view)
+
+        view.line_width = 1
+        
+        # Draw inputpoint (including tooltip) so the user knows where exactly
+        # where it is.
+        # However, don't draw it when at the selection stage and not hovering
+        # a selectable entity. Doing so would suggest you could click to select
+        # a point and proceed to the next tool stage.
+        if picked_first_point? || (@hovered_entity && @valid_hovered_entity)
+          @mouse_ip.draw(view)
+          view.tooltip = @mouse_ip.tooltip if @mouse_ip.valid?
+        end
+      end
+
+      def getExtents
+        bounds = Geom::BoundingBox.new
+        bounds.add(picked_points) unless picked_points.empty?
+        bounds
+      end
+
+      private
+
+      # Check if an entity is allowed to be selected by this tool.
+      def valid_entity?(entity)
+        # In this somewhat artificial example we only allow components to be
+        # moved. In a real life example you'd typically filter entities by
+        # other criteria.
+
+        # To replicate native Move, Rotate and Scale tools, allow all
+        # DrawingElements except Axes.
+        ### !entity.is_a?(Sketchup::Axes)
+
+        # Another typical use case is a custom tool that edits some property of
+        # a custom objects, e.g. Bezier curve control points or wall corner
+        # points. For this, you'd typically check for the entity being a group
+        # and/or component, and if it has the attribute dictionary holding your
+        # custom properties.
+        ### entity.is_a?(Sketchup::Group) && entity.attribute_dictionary("TurboWall2000")
+
+        entity.is_a?(Sketchup::ComponentInstance)
+      end
+
+      # Check if the selection as a whole is valid for this tool.
+      def valid_selection?
+        selection = Sketchup.active_model.selection
+
+        selection.size == 1 && valid_entity?(selection.first)
+      end
+
+      # Try selecting the entity under the mouse cursor, if the tool allows it.
+      def try_select_entity(view, x, y)
+        view.model.selection.clear
+
+        pickhelper = view.pick_helper(x, y)
+        @hovered_entity = pickhelper.best_picked
+        return unless @hovered_entity
+
+        @valid_hovered_entity = valid_entity?(@hovered_entity)
+        return unless @valid_hovered_entity
+
+        view.model.selection.add(@hovered_entity)
+      end
+
+      def start_move(view)
+        # HACK: create a new input point with a given position rather than
+        # copying the existing one. This fixes an issue in some cases where
+        # the points appear to get linked and the @picked_first_ip position
```

Comment:
"linked"?

Reply:
I don't know how to describe this bug. With the original code the reference input point countinosuly copy the value from the point that is being picked, causing the preview line to disappear and the VCB to show values close to 0. There is however a small delay, maybe from the previous pick, and the two InputPoints don't get the exact same position.

I don't know if it's a memory issue where both InputPoints get mixed up or what's causing it. This should be investigated, but probably not blocking the release of the UX guidelines.

Reply:
Yea, if you could capture the steps to reproduce, along with a couple of screenshots to illustrate what happens into a bug report in our issue tracker that would be good. I'm curious to see what is doing on here.

Reply:
I think this might be a quite deep rabbit hole but I make a simple issue that references the code example, so this work doesn't block publishing the UX Guidelines.

Reply:
Found it! (https://github.com/SketchUp/api-issue-tracker/issues/618)

Rewritten the comment to better communicate what is done and why.